tests/unit/select_connection_ioloop_tests.py

```diff
@@ -252,39 +293,140 @@ def on_timeout(self):
         raise AssertionError('Test timed out')
 
 class IOLoopSocketBasePoll(IOLoopSocketBaseSelect):
+    ''' Same as IOLoopSocketBaseSelect but uses 'poll' syscall '''
     SELECT_POLLER = 'poll'
 
 class IOLoopSocketBaseEPoll(IOLoopSocketBaseSelect):
+    ''' Same as IOLoopSocketBaseSelect but uses 'epoll' syscall '''
     SELECT_POLLER = 'epoll'
 
 class IOLoopSocketBaseKqueue(IOLoopSocketBaseSelect):
+    ''' Same as IOLoopSocketBaseSelect but uses 'kqueue' syscall '''
     SELECT_POLLER = 'kqueue'
 
 
 class IOLoopSimpleMessageTestCaseSelect(IOLoopSocketBaseSelect):
-
+    ''' Test read/write by creating a pair of sockets, writing to one
+        end and reading from the other '''
     def start(self):
+        ''' Create a pair of sockets and poll '''
         self.create_write_socket(self.connected)
         super(IOLoopSimpleMessageTestCaseSelect, self).start()
 
     def connected(self, fd, events, write_only):
+        ''' Respond to 'connected' event by writing to the write-side. '''
         self.assertEqual(events, WRITE)
+        self.assertTrue(write_only is not None)
         logging.debug("Writing to %d message: %s", fd, 'X')
         os.write(fd, b'X')
         self.ioloop.update_handler(fd, 0)
 
     def verify_message(self, msg):
+        ''' Make sure we get what is expected and stop polling '''
         self.assertEqual(msg, b'X')
         self.ioloop.stop()
 
     def start_test(self):
+        ''' Simple message Test'''
         self.start()
 
 class IOLoopSimpleMessageTestCasetPoll(IOLoopSimpleMessageTestCaseSelect):
+    ''' Same as IOLoopSimpleMessageTestCaseSelect but uses 'poll' syscall '''
     SELECT_POLLER = 'poll'
 
 class IOLoopSimpleMessageTestCasetEPoll(IOLoopSimpleMessageTestCaseSelect):
+    ''' Same as IOLoopSimpleMessageTestCaseSelect but uses 'epoll' syscall '''
     SELECT_POLLER = 'epoll'
 
 class IOLoopSimpleMessageTestCasetKqueue(IOLoopSimpleMessageTestCaseSelect):
+    ''' Same as IOLoopSimpleMessageTestCaseSelect but uses 'kqueue' syscall '''
+    SELECT_POLLER = 'kqueue'
+
+
+class DoAfterDelayThread(threading.Thread):
+    '''A class to send signal.SIGUSR1 to cause EINTR in reading thread.'''
+    def __init__(self, initial_wait, func, *args, **kw):
+        self.initial_wait = initial_wait
+        self.func = func
+        super(DoAfterDelayThread, self).__init__(*args, **kw)
+
+    def run(self):
+        time.sleep(self.initial_wait)
+        self.func()
+
+class IOLoopEintrTestCaseSelect(IOLoopBaseTest):
```

Comment:
Why derive from `IOLoopBaseTest`? This class is not using any of the core functionality from `IOLoopBaseTest` except the value of TIMEOUT, so it's not a "IOLoopBaseTest". Just derive from `unittest.TestCase` and reference `IOLoopBaseTest.TIMEOUT` where you need to use the same TIMEOUT.